src/pandas_openscm/index_manipulation.py

```diff
@@ -718,7 +718,7 @@ def update_levels_from_other(
 
 def create_level_from_collection(
     level: str, value: Collection[Any]
-) -> tuple[pandas.Index[Any], list[int]]:
+) -> tuple[pandas.Index[Any], npt.NDArray[np.integer[Any]]]:
```

Comment:
The second return value is the codes, so should have the same type as we expect codes to have